tests/test_requests.py

```diff
@@ -606,6 +626,27 @@ def test_unicode_header_name(self, httpbin):
     def test_pyopenssl_redirect(self, httpbin_secure, httpbin_ca_bundle):
         requests.get(httpbin_secure('status', '301'), verify=httpbin_ca_bundle)
 
+    def test_https_warnings(self, httpbin_secure, httpbin_ca_bundle):
+        """warnings are emitted with requests.get"""
+        if HAS_MODERN_SSL or HAS_PYOPENSSL:
+            warnings_expected = ('SubjectAltNameWarning', )
```

Comment:
I have one concern about the patch (and series in general), and that is whether this warning should appear every time, or only the first time for a connection , and is the connection to httpbin_secure being cached.
My reading of VerifiedHTTPSConnection suggests it should only happen once per connection, and I am not sure if httpbin_secure is cached.